pkg/actors/actors.go

```diff
@@ -538,11 +569,71 @@ func (a *actorsRuntime) callRemoteActor(
 	return invokeResponse, nil
 }
 
+func (a *actorsRuntime) callInternalActor(ctx context.Context, req *invokev1.InvokeMethodRequest) (*invokev1.InvokeMethodResponse, error) {
+	actorTypeID := req.Actor()
+	act := a.getOrCreateActor(actorTypeID.GetActorType(), actorTypeID.GetActorId())
+	factory, ok := a.internalActors[act.actorType]
+	if !ok {
+		return nil, fmt.Errorf("internal actor type '%s' is not registered", act.actorType)
+	}
+
+	// The factory is expected to manage its own internal caching of active actor instances
+	impl := factory(a)
+
+	// Internal actors don't support any form of reentrancy (for now)
+	reentrancyID := uuid.New().String()
+	if err := act.lock(&reentrancyID); err != nil {
+		return nil, status.Error(codes.ResourceExhausted, err.Error())
+	}
+	defer act.unlock()
+
+	methodName := req.Message().GetMethod()
+	requestData := req.Message().GetData().GetValue()
+	verb := req.Message().GetHttpExtension().GetVerb()
+
+	// Call the appropriate method based on the method name and verb.
+	var result interface{} = nil
+	var err error = nil
+	if strings.HasPrefix(methodName, "remind/") {
+		reminderName := strings.TrimPrefix(methodName, "remind/")
+		err = impl.InvokeReminder(ctx, act.actorID, reminderName, requestData)
+	} else if strings.HasPrefix(methodName, "timer/") {
+		timerName := strings.TrimPrefix(methodName, "timer/")
+		err = impl.InvokeTimer(ctx, act.actorID, timerName, requestData)
+	} else if verb == commonv1pb.HTTPExtension_DELETE {
```

Comment:
I'm not sure what to do with the linter warning I'm getting here. It says the following:

> HTTPExtension_DELETE contains underscore. You should use mixedCap or MixedCap. (nosnakecase).

However, this is generated code and I'm not actually declaring it here, just referencing it. I can't change the declaration and I don't see a way to suppress this.

I could maybe do something like this:

```go
} else if int(verb) == 5 /* DELETE */ {
```

This seems like a bad idea for obvious reasons. Any suggestions? I assume this will block the PR from being merged.

Reply:
Verb is already an int32 I think -- if so you should be able to define a constant up top like `const HTTPDelete = 5` and then just say `if (verb == HTTPDelete)` (or whatever you want to call the constant)

Reply:
I figured out that I can use a `//nolint:nosnakecase` comment to suppress this linter error (found it used elsewhere in the dapr/dapr code for this same type), so I was able to keep the original code and make the error go away.